tests/ffi/memory/prepare.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+# shellcheck disable=SC1091
+
+
+prepare_test() {
+
+   qm_service_file=$(systemctl show -P  SourcePath qm)
+
+   #create backup file for qm unit file
+   qm_service_backup=$(mktemp -d -p /tmp)/qm.service
+   cp "${qm_service_file}" "${qm_service_backup}"
```

Comment:
I would check if the command worked via $?  or use a function to reuse the validation, such:
```
ec_cmd() {
    local cmd="$1"
    eval "$cmd"
    local exit_code="$?"

    if [ "$exit_code" -ne 0 ]; then
        echo "Error: Command failed with exit code $exit_code"
        exit "$exit_code"
    fi
}
```

Reply:
Ack